### PR TITLE
feat: add extensions support

### PR DESCRIPTION
**Pull request by sambhavgupta0705**
This PR is part of asyncapi mentorship program
related issue: https://github.com/asyncapi/spec-json-schemas/issues/280

The previous was targeted to `next-major-spec` branch
All the changes have been copied and migrated to master branch

### CODE REVIEW

**Comment by sambhavgupta0705** [comment]
cc: @derberg  @smoya

**Comment by sambhavgupta0705** [comment]
@derberg the previous pr was targetting to `next-spec-major` and this is targetting `master`.
I just copied the changes directly in this and now there are some deletions in https://github.com/asyncapi/spec-json-schemas/pull/506/files#diff-6deb33116f8f13032612ff1f8a6305914860872230d74fa630e77e3a277be4a2 for which I would like to know what should I do.

- should I only add extension related changes
- or should let this file as it is

**Comment by derberg** [comment]
@sambhavgupta0705 sorry can you point to some example line of code and give more context, I'm not sure I follow

File: tools/bundler/index.js [hunk 1]
```diff
@@ -1,27 +1,32 @@
 const path = require('path');
 const fs = require('fs');
 const traverse = require('json-schema-traverse');
+const { url } = require('inspector');
```
Comment (sambhavgupta0705) [inline tools/bundler/index.js:4]: @derberg for example this
This wasn't in your pr but was added in `next-major-spec`

Comment (derberg) [reply tools/bundler/index.js:4]: I don't see it being used in code

Comment (sambhavgupta0705) [reply tools/bundler/index.js:4]: Yes 
When you raised a pr on my branch then it was there but after that it was removed from master

**Review by derberg — APPROVED** [review]
Approving and merging as it is a PR that I worked with @sambhavgupta0705 for quite some time already, https://github.com/asyncapi/spec-json-schemas/pull/443/ was just merged into a wring branch 😄

**Pull request merged**